src/OrchardCore/OrchardCore.Navigation.Core/NavigationHelper.cs

```diff
@@ -68,11 +66,17 @@ private static async Task<dynamic> BuildMenuItemShapeAsync(dynamic shapeFactory,
                 .Level(parentShape.Level == null ? 1 : (int)parentShape.Level + 1)
                 .Priority(menuItem.Priority)
                 .Local(menuItem.LocalNav)
+                .Hash((parentShape.Hash + menuItem.Text.Value).GetHashCode().ToString())
                 .Score(0);
 
             menuItemShape.Id = menuItem.Id;
 
-            MarkAsSelectedIfMatchesRouteOrUrl(menuItem, menuItemShape, viewContext);
+            if (menuItem.Href?[0] == '/')
```

Comment:
Can it be empty and fail?

Reply:
i think i checked it if i wrote it like this, but not sure, will check it again

Reply:
It is computed in `NavigationManager.ComputeHref`, for link node it is at least `#`, then for a route values based menu item i tried it with an empty route values dictionary and i got `/admin` (based on the current request path), so i think it would be at least `/`.

So at least it is equal to `#` or `/` but not totally sure ;) so i will add the null checks.

Reply:
A PlaceholderAdminMenuLink ?

Reply:
In this case href is set to `#` by `ComputeHref()` => `GetUrl`

Reply:
okay ! don't break anything 😄